src/FastMathFunctions.cpp

```diff
@@ -0,0 +1,769 @@
+#include "FastMathFunctions.h"
+
+#include "ApproximationTables.h"
+#include "CSE.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "IRPrinter.h"
+
+namespace Halide {
+namespace Internal {
+
+// Implemented in IROperator.cpp
+void range_reduce_log(const Expr &input, Expr *reduced, Expr *exponent);
```

Comment:
Put this in a header somewhere.

Reply:
`IROperator.h` is public API. It feels wrong to put it there. That's why I did this for now. What's a good place to put this, you think?

Reply:
Anything inside the `Halide::Internal` namespace is "use at your own risk".